molecule/driver/delegated.py

```diff
@@ -151,20 +151,33 @@ def ansible_connection_options(self, instance_name):
             try:
                 d = self._get_instance_config(instance_name)
 
-                return {
-                    'ansible_user':
-                    d['user'],
-                    'ansible_host':
-                    d['address'],
-                    'ansible_port':
-                    d['port'],
-                    'ansible_private_key_file':
-                    d['identity_file'],
-                    'connection':
-                    'ssh',
-                    'ansible_ssh_common_args':
-                    ' '.join(self.ssh_connection_options),
-                }
+                if d['connection'] == "winrm":
```

Comment:
Hmmm, this doesn't seem like the right change. Do we really need to delve directly into the code here just to allow this new type of connection? Is there some more general way we can allow this customisation? This strategy won't work for further kinds of `d['connection']` which will make the code hard to maintain.

Reply:
@lwm these temporary solution, we do some tests now for improvement.